content/doc/tutorials/using-jenkinsfile-runner-github-action-to-build-jenkins-pipeline.adoc

```diff
@@ -0,0 +1,308 @@
+* Specify `adoptopenjdk` plugin in the plugins.txt file. 
+As the version is not specified, the latest version will be installed.
+[source,txt]
+----
+adoptopenjdk
+----
+* Create a JCasC Yaml file called jcasc.yml and specify which JDK version to install.
+[source,yaml]
+----
+tool:
+  jdk:
+    installations:
+      - name: jdk8
+        home: "~/jdk8"
+        properties:
+          - installSource:
+              installers:
+                - adoptOpenJdkInstaller:
+                    id: "jdk8u332-b09"
+----
+* Specify the jcasc.yml in the GitHub Actions input.
+[source,yaml]
+----
+uses:
+  jenkinsci/jfr-container-action@master
+with:
+  command: run
+  jenkinsfile: Jenkinsfile
+  pluginstxt: plugins.txt
+  jcasc: jcasc.yml
+----
+
+=== Configure ephemeral Jenkins instance (Optional)
+
+Sometimes JCasC might not be able to provide the configurations you need. 
+In this case, you can use link:https://www.jenkins.io/doc/book/managing/groovy-hook-scripts/[Groovy Hook Scripts] to set up the ephemeral Jenkins instance. 
+These Groovy scripts will have full access to the ephemeral Jenkins server and will be executed right after Jenkins starts up. 
+Please note this option and its core are still in progress so it’s not mentioned in the Jenkinsfile Runner GitHub Actions official guide. 
+But it does work.
+
+In this example, we show how to use the Groovy script to set up the Jenkins instance:
+* Create a directory called “groovy.init.d” (this name can be something else) to store all your Groovy setup scripts.
+* Create a Groovy file called “test.groovy”. (Please don’t name it as init.groovy because this name is already occupied.) Then, add the debug output:
+[source,groovy]
+----
+println 'Hello Groovy Hooks!'
+----
+* Specify the “groovy.init.d” directory in the GitHub Actions input.
+[source,yaml]
+----
+uses:
+  jenkinsci/jfr-container-action@master
+with:
+  command: run
+  jenkinsfile: Jenkinsfile
+  pluginstxt: plugins.txt
+  jcasc: jcasc.yml
+  initHook: groovy.init.d
+----
+* Check the GitHub Actions log and you can find the groovy script is executed right after Jenkins starts up and before the actual pipeline.
+
+=== Integrate with other GitHub Actions (Optional)
+
+You can integrate this with other GitHub Actions in the marketplace via some Jenkinsfile Runner GitHub Actions. 
+However, as the starting time of Jenkins container is different in these GitHub Actions, some GitHub Actions in the marketplace cannot be used. 
+In other words, `jfr-static-image-action` cannot be integrated with the other GitHub Actions except `actions/checkout` but you can integrate other GitHub Actions with `jfr-container-action` and `jfr-runtime-action`. 
+Feel free to check out link:https://jenkinsci.github.io/jfr-action-doc/docs/user-guide/actions-cmp.html[their differences in the official guide].
+
+In this example, we show how to integrate actions/setup-node GitHub Action with jfr-runtime-action to compile a JavaScript project:
```

Comment:
```suggestion
In this example, we describe how to integrate GitHub actions with jfr-runtime-action to compile a JavaScript project:
```
syntax suggestion, if this takes away from the meaning, disregard

Reply:
In this example, I think we need to specify other special GitHub Actions in the GitHub Actions marketplace.